### PR TITLE
support for Ubuntu 22.04 in compile scripts and system includes

### PR DESCRIPTION
**Pull request by NeptuneMagicSauce**

### CODE REVIEW

File: build/scripts/build_helper.fb_folly [hunk 1]
```diff
@@ -132,6 +132,10 @@ install_fb_folly_from_source(){
     else
         git checkout -f v2019.11.11.00
     fi
+    if [[ "$(get_distribution_release)" == "ubuntu22.04" ]]; then
+       sed -i -e "s@#include <algorithm>@#include <algorithm>\n#include <limits>@" folly/TimeoutQueue.cpp
+       sed -i -e "s@folly::constexpr_max@@" -e "s@SIGSTKSZ, @@" ./folly/fibers/FiberManager.cpp
+    fi
```
Comment (rdefosse) [inline build/scripts/build_helper.fb_folly:138]: @NeptuneMagicSauce (romain)

I would do this patch all the time. I know we have to do it also for `fedora-38`.

**Review by rdefosse — CHANGES_REQUESTED** [review]
Please

File: build/scripts/build_helper [hunk 2]
```diff
@@ -129,6 +129,7 @@ check_supported_distribution() {
     case "$distribution" in
         "ubuntu18.04") return 0 ;;
         "ubuntu20.04") return 0 ;;
+        "ubuntu22.04") return 0 ;;
```
Comment (rdefosse) [inline build/scripts/build_helper:132]: @NeptuneMagicSauce 

Add also "rhel8.7" at the end of the list. That's the reason it failed.

**Review by rdefosse — CHANGES_REQUESTED** [review]

File: build/scripts/build_helper (hunk 2 above)
Comment (rdefosse) [reply build/scripts/build_helper:132]: After `yum update`:

```bash
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.7 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.7"
...
```

File: build/scripts/build_helper.fb_folly (hunk 1 above)
Comment (NeptuneMagicSauce) [reply build/scripts/build_helper.fb_folly:138]: Will this be backward-compatible? Will it work for Ubuntus 18 and 20?

Comment (rdefosse) [reply build/scripts/build_helper.fb_folly:138]: yes

**Comment by rdefosse** [comment]
@NeptuneMagicSauce @arora-sagar 

We have problem: the build occured on `orion` but the 4G test occured on `selfix`.

Just kidding. I really want to have the `local registry` in the CN (at least for our own components not necessarily for MAGMA)

**Comment by rdefosse** [comment]
@NeptuneMagicSauce 

Let's put this MR in hold. We need to resolve some CI issues.

**Pull request merged**